homeassistant/components/airios/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The Airios integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pyairios import Airios, AiriosRtuTransport, AiriosTcpTransport
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_TYPE,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DEFAULT_SCAN_INTERVAL, BridgeType
+from .coordinator import AiriosDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [
+    Platform.FAN,
+]
+
+
+type AiriosConfigEntry = ConfigEntry[AiriosDataUpdateCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: AiriosConfigEntry) -> bool:
+    """Set up Airios from a config entry."""
+
+    bridge_type = entry.data[CONF_TYPE]
+    if bridge_type == BridgeType.SERIAL:
+        device = entry.data[CONF_DEVICE]
+        transport = AiriosRtuTransport(device)
+    elif bridge_type == BridgeType.NETWORK:
+        host = entry.data[CONF_HOST]
+        port = entry.data[CONF_PORT]
+        transport = AiriosTcpTransport(host, port)
+    else:
+        raise ConfigEntryNotReady(f"Unexpected bridge type {bridge_type}")
+
+    modbus_address = entry.data[CONF_ADDRESS]
+    api = Airios(transport, modbus_address)
+
+    update_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    coordinator = AiriosDataUpdateCoordinator(hass, api, update_interval)
+    await coordinator.async_config_entry_first_refresh()
+
+    bridge_rf_address = await api.bridge.node_rf_address()
+    if bridge_rf_address is None or bridge_rf_address.value is None:
+        raise ConfigEntryNotReady("Failed to get bridge RF address")
+    bridge_rf_address = bridge_rf_address.value
+
+    if entry.unique_id != str(bridge_rf_address):
+        message = (
+            f"Unexpected device {bridge_rf_address} found, expected {entry.unique_id}"
+        )
+        _LOGGER.error(message)
+        raise ConfigEntryNotReady(message)
```

Comment:
How does this work?

Reply:
Each node in a RAMSES-II network has an RF address (0xAABBCC), usually used as the serial number. The bridge is part of the network so it has its own RF address too. The other nodes, like ventilation units or remotes bind to the bridge. The bridge creates a virtual Modbus unit with its own Modbus address for each bound node with its own register set. This particular check verifies that the connected bridge is the expected one by verifying its RF address (or serial number).